app/questionnaire/questionnaire_schema.py

```diff
@@ -907,23 +913,29 @@ def get_questions(self, question_id: str) -> list[ImmutableDict] | None:
         return self._questions_by_id.get(question_id)
 
     def get_list_collectors_for_list(
-        self, section: ImmutableDict, for_list: str, primary: bool = False
-    ) -> Generator[ImmutableDict, None, None]:
-        collector_type = "PrimaryPersonListCollector" if primary else "ListCollector"
+        self, sections: list, for_list: str, primary: bool = False
+    ) -> list:
+        blocks: list = []
+        for section_id in sections:
+            if section := self.get_section(section_id):
+                collector_type = (
+                    "PrimaryPersonListCollector" if primary else "ListCollector"
+                )
 
-        return (
-            block
-            for block in self.get_blocks_for_section(section)
-            if block["type"] == collector_type and block["for_list"] == for_list
-        )
+                blocks.extend(
+                    block
+                    for block in self.get_blocks_for_section(section)
+                    if block["type"] == collector_type and block["for_list"] == for_list
+                )
+
+        return blocks
 
     def get_list_collector_for_list(
         self, for_list: str, primary: bool = False
-    ) -> ImmutableDict | None:
-        # Type ignore: Section will exist at this point
-        section: ImmutableDict = self.get_section(self._list_name_to_section_id_origin_map[for_list])  # type: ignore
+    ) -> list[ImmutableDict]:
+        sections = self._list_name_to_section_id_origin_map[for_list]
 
-        return next(self.get_list_collectors_for_list(section, for_list, primary))
+        return self.get_list_collectors_for_list(sections, for_list, primary)
```

Comment:
This name needs changes as the above comment will clash with this. This can be `get_list_collectors_for_list_for_sections`.

Reply:
Updated 👍